internal/scorecard/alpha/tests/olm.go

```diff
@@ -83,3 +107,77 @@ func StatusDescriptorsTest(bundle registry.Bundle) scapiv1alpha2.ScorecardTestRe
 	r.Suggestions = make([]string, 0)
 	return r
 }
+
+func validateCR(cr unstructured.Unstructured, crds []*v1beta1.CustomResourceDefinition,
+	r scapiv1alpha2.ScorecardTestResult) scapiv1alpha2.ScorecardTestResult {
+
+	// check if the CRD matches the testing CR
+	for _, crd := range crds {
+		gvk := cr.GroupVersionKind()
+		// Only check the validation block if the CRD and CR have the same Kind and Version
+		if !(matchVersion(gvk.Version, crd) && matchKind(gvk.Kind, crd.Spec.Names.Kind)) {
+			continue
```

Comment:
I'd export it to a func to make more clear what it does and do such as follows instead of the above if to try make easier get what it is doing.

```
if cr.isFromCRD(crd) {
    // code 
}
```

Reply:
I  believe `matchVersion` and `matchKind` are doing more comprehensive comparison, than what you are suggesting above. `matchKind` is using `ResourceSingulaizer`. And as i see `matchVersion` is looping through multiple versions if exists. I can change the names as you have suggested to `isMatchVersion` and `isMatchKind`.